app/src/main/java/com/ably/tracking/demo/publisher/domain/OrderManager.kt

```diff
@@ -0,0 +1,130 @@
+package com.ably.tracking.demo.publisher.domain
+
+import com.ably.tracking.TrackableState
+import com.ably.tracking.demo.publisher.ably.AssetTracker
+import com.ably.tracking.demo.publisher.api.DeliveryServiceDataSource
+import com.ably.tracking.demo.publisher.common.copyAndReplaceElementAt
+import com.ably.tracking.demo.publisher.common.toOrderState
+import com.ably.tracking.publisher.Trackable
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+interface OrderManager {
+    val orders: MutableStateFlow<List<Order>>
```

Comment:
Is there a reason why it's not an immutable `StateFlow`? :thinking:

Reply:
b5f8ba181aedd0a8aeff517733cfa2fe2d6dc992